### DISCUSSION TITLE
Fail behaviour of `klog switch` / `klog start`

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/klog#287] [Feature ideas & discussions]
This is a spin-off discussion of https://github.com/jotaen/klog/discussions/285.

`klog switch` currently fails if there is no open range in the record. It could be argued, that `klog switch` could just fall back to behaving like `klog start` if there is no open range in the record, mostly for convenience.

A similar point could be made about `klog start`, which currently fails if there is already an open range in the record. But one could say that `klog start` could behave like `klog switch` in this case, again for convenience.

Back then, I’ve deliberately made it fail, because I was thinking the inconvenience of it failing explicitly would be better than having some potentially unexpected, implicit default behaviour that may lead to wrong data. Also, because if both commands would automatically fall back to behaving like the other respective command, then why would there be two separate commands to begin with?

One potential approach could be to leave the current behaviour as is, but add some sort of `--force` option. I’d only find that sensible for `klog switch`, though. It also may be useful to print a warning if the precondition isn’t met.

**Comment by lakemike** [comment]
Voting +1 for having a "--force" flag added to the switch command. :)

Having such a command would be convenient in my use case, because I call klog via iOS shortcuts "Run script over ssh".
Not having such a command: Either I see the iOS shortcut failing and as a user I need to respond with a follow-up action. Or I need to add some scripting at the linux server (if-then) to add this feature on my own.
